src/functional/fencil_processors/runners/roundtrip.py

```diff
@@ -179,9 +176,9 @@ def register_tmp(tmp, domain):
         if not debug:
             pathlib.Path(source_file_name).unlink(missing_ok=True)
 
-    assert isinstance(ir, FencilDefinition)
-    fencil_name = ir.id
-    fencil = getattr(foo, fencil_name + "_wrapper")
+    assert isinstance(ir, (itir.FencilDefinition, FencilWithTemporaries))
+    fencil_name = ir.fencil.id + "_wrapper" if isinstance(ir, FencilWithTemporaries) else ir.id
+    fencil = getattr(foo, fencil_name)
```

Comment:
We should rename foo. I guess I approved it in the previous commit 😳

Reply:
Fixed you terrible review failure 😂